juju/juju
# Lease expiry domain

In addition to the lease domain PR (#15807) the following moves over the expiry part of the equation to the lease domain. This keeps everything in one location for leases.

The majority of the work was just pushing the store through the worker scaffolding and placing the sql statements in the lease domain state.

Also, this cleans up the worker nicely, as we're separating the concerns of when it should be run and what should be run.

## Checklist



## QA steps

```sh
$ juju bootstrap lxd test --build-agent
$ juju add-model default
$ juju enable-ha
$ juju deploy ubuntu -n 3
```
Ensure that there are no lease errors.
